cinn/frontend/pass/transpose_collapsing.cc

```diff
@@ -169,27 +170,49 @@ class TransposeCollapsingPass : public ProgramPass {
         << "The transpose's axis size should equal with input variable's shape size, but the transpose of ["
         << input->id << "] not ! Please check.";
 
+    bool can_remove = !fetch_ids.count(output_name);
+
     if (CheckTransposeBorder(transpose, in2instr)) {
-      VLOG(4) << "The transpose op {input[" << input_name << "], output[" << output_name << "], axis["
-              << cinn::utils::Join(axis, ",") << "]} is a output op of graph, connot fuse, skip.";
+      if (can_remove) {
+        VLOG(4) << "The transpose op {input[" << input_name << "], output[" << output_name << "], axis["
+                << cinn::utils::Join(axis, ",") << "]} is a output op of graph, connot fuse, remove.";
+        // this transpose not used by any other op, remove
+        remove_instrs->insert(transpose);
+      } else {
+        if (CheckTransposeUseless(axis)) {
+          VLOG(4) << "The transpose op {input[" << input_name << "], output[" << output_name << "], axis["
+                  << cinn::utils::Join(axis, ",") << "]} is fetched but useless, replace with identity.";
+          // cannot remove, however, the transpsoe is useless, we can replace the transpose with indentiy for more
```

Comment:
why cannot remove it?

Reply:
because  its`output_name` in `fetch_ids`